src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/ConfigModel/TemplateConfigModel.cs

```diff
@@ -507,7 +515,358 @@ private static IReadOnlyList<BindSymbol> SetupImplicitBindSymbols()
+                                    writer.WritePropertyName("copyOnly");
+                                    if (mod.CopyOnly.Count == 1)
+                                    {
+                                        writer.WriteValue(mod.CopyOnly[0]);
+                                    }
+                                    else
+                                    {
+                                        writer.WriteStartArray();
+                                        foreach (string el in mod.CopyOnly)
+                                        {
+                                            writer.WriteValue(el);
+                                        }
+                                        writer.WriteEndArray();
+                                    }
+                                }
+                                if (mod.Rename.Any())
+                                {
+                                    writer.WritePropertyName("rename");
+                                    foreach (KeyValuePair<string, string> el in mod.Rename)
+                                    {
+                                        writer.WriteStartObject();
+                                        writer.WritePropertyName(el.Key);
+                                        writer.WriteValue(el.Value);
+                                        writer.WriteEndObject();
+                                    }
+                                }
+                                writer.WriteEndObject();
+                            }
+                            writer.WriteEndArray();
+                        }
+                        writer.WriteEndObject();
+                    }
+                    writer.WriteEndArray();
+                }
+
+                //not implemented
+                if (value.Forms.Any())
+                {
+                    throw new NotSupportedException("Forms are not supported for serialization to JSON.");
+                }
+                if (value.BaselineInfo.Any())
+                {
+                    throw new NotSupportedException("Baselines are not supported for serialization to JSON.");
+                }
+                if (value.Symbols.Any(s => !s.IsImplicit))
+                {
+                    writer.WritePropertyName("symbols");
+                    writer.WriteStartObject();
+                    foreach (ParameterSymbol p in value.Symbols.OfType<ParameterSymbol>())
+                    {
+                        if (p.IsImplicit)
+                        {
+                            continue;
+                        }
+                        serializer.Serialize(writer, p);
+                    }
+                    writer.WriteEndObject();
+
+                    if (value.Symbols.Any(s => s is not ParameterSymbol && !s.IsImplicit))
+                    {
+                        throw new NotSupportedException("Symbols are not supported for serialization to JSON.");
```

Comment:
Message seams bit missleading. Maybe "Only Implicit and Parameter symbols are supported for serialization to JSON."?

Reply:
Just wondering - why are the other types not supported? Will wee need the serializer when validating e.g. template with calculated symbols?

Reply:
It just needs more work and not required for time being. I'll create an issue for this. Also I will try to move this serialization to tests not to misuse unintentionally.

Reply:
Moved serialization to tests as it is not complete and can be used only for tests.
The idea is that if the next tests need more data, the serialization may be implemented then.